sw/host/opentitanlib/src/uart/console.rs

```diff
@@ -87,10 +87,14 @@ impl UartConsole {
                     .map_or(Ok(()), |out| out.write_fmt(format_args!("[{}]", t)))?;
                 self.newline = false;
             }
-            stdout
-                .as_mut()
-                .map_or(Ok(()), |out| out.write_all(&buf[i..i + 1]))?;
             self.newline = buf[i] == b'\n';
+            stdout.as_mut().map_or(Ok(()), |out| {
```

Comment:
nit: can add an extra `CR` if the line already ends in `CRLF` which can cause confusion if we decide to save opentitantool's output to a file.

Reply:
Created #14591.